libs/ballot-interpreter/src/hmpb-rust/timing_marks.rs

```diff
@@ -1538,17 +1545,22 @@ fn infer_missing_timing_marks_on_segment(
         return vec![];
     }
 
+    let segment_forward = segment;
+    let segment_backward = segment_forward.reversed();
+
     let mut inferred_timing_marks = vec![];
-    let mut current_timing_mark_center = segment.start;
-    let next_point_vector = segment.with_length(expected_distance).vector();
+    let mut current_timing_mark_center_forward = segment_forward.start;
+    let mut current_timing_mark_center_backward = segment_backward.start;
+    let next_point_vector_forward = segment_forward.with_length(expected_distance).vector();
+    let next_point_vector_backward = segment_backward.with_length(expected_distance).vector();
     let maximum_error = expected_distance / 2.0;
```

Comment:
How should `maximum_error` here relate to the max gap allowed in the resulting timing marks?

Reply:
Consolidated with the other change.